### PR TITLE
[WIP] weekly-retraining-correction-v3

### PR DESCRIPTION
**Pull request by anuragtr**
separate function for weekly re-training created
collated-monthly.json also taken care of
tested on dev cluster

### CODE REVIEW

File: f8a_report/main.py [hunk 1]
```diff
@@ -22,18 +22,24 @@ def main():
 
     start_date = (today - timedelta(days=1)).strftime('%Y-%m-%d')
     end_date = today.strftime('%Y-%m-%d')
-    response, ingestion_results = r.get_report(start_date, end_date, 'daily')
+    response, ingestion_results = r.get_report(start_date, end_date, 'daily', retrain='F')
```
Comment (miteshvp) [inline f8a_report/main.py]: It's always better to use boolean values `retrain=False`. And everywhere else.

File: f8a_report/report_helper.py [hunk 2]
```diff
@@ -404,18 +405,16 @@ def normalize_worker_data(self, start_date, end_date, stack_data, worker, freque
...
-            if today.weekday() >= 0:
-                # Collate Data from Previous Month for Model Retraining
+            if retrain == 'T':
+                # Appends last week's data to 'collated-weekly'; returns 'BQ+collated' data
                 collated_data = self.collate_raw_data(unique_stacks_with_recurrence_count,
                                                       'weekly')
-                # Store ecosystem specific data to their respective Training Buckets
+                # Stores ecosystem specific manifest.json and re-trains models
                 self.store_training_data(collated_data)
+                return
```
Comment (miteshvp) [inline f8a_report/report_helper.py]: This return statement assumes that you will not do retraining and normal venus reporting together ever. It's either/or. Rather that that, you could have moved or broken this function down into two modules. Code before this if statement and after this if statement as two different functions. Then handle the if part accordingly.

File: f8a_report/report_helper.py [hunk 3]
```diff
@@ -743,9 +742,20 @@ def get_report(self, start_date, end_date, frequency='daily'):
...
             worker_result = self.retrieve_worker_results(
-                start_date, end_date, ids, ['stack_aggregator_v2'], frequency)
+                start_date, end_date, ids, ['stack_aggregator_v2'], frequency, retrain)
             return worker_result, ingestion_results
         else:
             logger.error('No stack analyses found from {s} to {e} to generate an aggregated report'
                          .format(s=start_date, e=end_date))
             return False, ingestion_results
+
+    def retrain(self, start_date, end_date, frequency='weekly', retrain='T'):
```
Comment (miteshvp) [inline f8a_report/report_helper.py]: Please add unit test cases for this new function

File: f8a_report/main.py (hunk 1 above)
Comment (anuragtr) [reply f8a_report/main.py]: ok sure, will do it

File: f8a_report/report_helper.py (hunk 3 above)
Comment (anuragtr) [reply f8a_report/report_helper.py]: yep adding

File: f8a_report/report_helper.py (hunk 2 above)
Comment (anuragtr) [reply f8a_report/report_helper.py]: @miteshvp as discussed, working on breaking the functions.

**Pull request closed without merging**